### PR TITLE
skip health checks for jobs, cronjobs, and initContainers

### LINKED ISSUE #215: Liveness probes should be ignored on initContainers
Adding Livenessprobe to InitContainers results in failures. Are these even supported by k8s?

### PR DESCRIPTION
**Pull request by rbren**
Fixes #215 and #209 

Also skips resource checks if they're not in the config

### CODE REVIEW

File: pkg/validator/container.go [hunk 1]
```diff
@@ -38,7 +38,7 @@ type ContainerValidation struct {
 //       relevant podSpec in order to check certain aspects of a containerSpec.
 //       Perhaps there is a more ideal solution instead of attaching a parent
 //       podSpec to every container Validation struct...
-func ValidateContainer(container *corev1.Container, parentPodResult *PodResult, controllerName string, conf *config.Configuration, isInit bool) ContainerResult {
+func ValidateContainer(container *corev1.Container, parentPodResult *PodResult, conf *config.Configuration, controllerName string, controllerType config.SupportedController, isInit bool) ContainerResult {
```
Comment (jordandoig) [inline pkg/validator/container.go]: More of a golang question here, is there a reason the argument order here needed to be changed?

File: pkg/validator/pod.go [hunk 2]
```diff
@@ -27,7 +27,7 @@ type PodValidation struct {
 }
 
 // ValidatePod validates that each pod conforms to the Polaris config, returns a ResourceResult.
-func ValidatePod(conf config.Configuration, controllerName string, pod *corev1.PodSpec) PodResult {
+func ValidatePod(conf config.Configuration, pod *corev1.PodSpec, controllerName string, controllerType config.SupportedController) PodResult {
```
Comment (jordandoig) [inline pkg/validator/pod.go]: Same as above, re: argument order

File: pkg/validator/container.go (hunk 1 above)
Comment (rbren) [reply pkg/validator/container.go]: Definitely wasn't necessary. I merged in the change that added the `controllerType` parameter, and realized `controllerType` and `controllerName` should probably be next to each other.

File: pkg/validator/pod.go (hunk 2 above)
Comment (rbren) [reply pkg/validator/pod.go]: Ditto - just an aesthetic preference

**Pull request merged**